### ISSUE TITLE
NaN handling of stats.rankdata

### ISSUE THREAD

**Issue by AtsushiSakai** [scipy/scipy#11790]
I think NaN handling of stats.rankdata is not correct. 

#### Reproducing code example:

Current code is

```
rankdata([0, 2, 3, 2, np.nan, np.nan])
> [1.  2.5 4.  2.5 5.  6. ]
```

However, I think it should be

```
rankdata([0, 2, 3, 2, np.nan, np.nan])
> [1.  2.5 4.  2.5 np.nan  np.nan ]
```

or 

return nan

or

raise an exception.

#### Scipy/Numpy/Python version information:
Python: 3.7.7
scipy: 1.7.1

**Comment by chrisb83** [comment]
see also the related issue #11409

**Comment by AtsushiSakai** [comment]
@chrisb83 Oh. Thanks. I will try to fix it in #11791
